test/integration_tests/src/test_latency_aware_policy.cpp

```diff
@@ -0,0 +1,245 @@
+
+  LatencyAwarePolicyTest()
+    : configuration_(cql::get_ccm_bridge_configuration())
+    , cluster_(cass_cluster_new())
+    , thread_()
+#if UV_VERSION_MAJOR == 0
+    , loop_(uv_loop_new()) {
+# else
+  {
+    boost::debug::detect_memory_leaks(false);
+    uv_loop_init(&loop_);
+#endif
+    // Initialize the cluster for latency aware
+    cass_cluster_set_reconnect_wait_time(cluster_.get(), 1);
+    cass_cluster_set_connect_timeout(cluster_.get(), 240 * test_utils::ONE_SECOND_IN_MICROS);
+    cass_cluster_set_request_timeout(cluster_.get(), 240 * test_utils::ONE_SECOND_IN_MICROS);
+    test_utils::initialize_contact_points(cluster_.get(), configuration_.ip_prefix(), 3, 0);
+    cass_cluster_set_latency_aware_routing(cluster_.get(), cass_true);
+    cass_cluster_set_latency_aware_routing_settings(cluster_.get(), 1e6, 1, 1, 1, 1);
+    cass_cluster_set_protocol_version(cluster_.get(), 1); // Protocol for this test doesn't matter so simply support all C* versions
+
+    // Create the cluster and connect
+    ccm_ = cql::cql_ccm_bridge_t::create_and_start(configuration_, "test", 3, 0);
+    session_ = test_utils::create_session(cluster_.get());
+  }
+
+  ~LatencyAwarePolicyTest() {
+    stop_query_execution();
+#if UV_VERSION_MAJOR == 0
+    uv_loop_delete(loop_);
+    loop - NULL;
+#else
+    uv_loop_close(&loop_);
+#endif
+
+    test_utils::CassFuturePtr close_future(cass_session_close(session_.get()));
+    cass_future_wait(close_future.get());
+  }
+
+  /**
+   * Start the query execution thread
+   */
+  void start_query_execution() {
+    uv_thread_create(&thread_, start_thread, NULL);
+  }
+
+  /**
+   * Stop the executing query thread
+   */
+  void stop_query_execution() {
+    is_running_ = false;
+    uv_thread_join(&thread_);
+  }
+
+  /**
+   * Create latency on a node using CCM to pause/resume execution of the node
+   *
+   * @param node Node to create latency on
+   * @param latency Latency to incur
+   */
+  void create_latency(int node, unsigned int latency) {
+    // Add latency to the node
+    ccm_->pause(node);
+    boost::this_thread::sleep_for(boost::chrono::milliseconds(latency));
+    ccm_->resume(node);
+  }
+
+  /**
+   * Check the maximum latency incurred on a node and ensure that is it within
+   * accepted threshold/tolerance
+   *
+   * @param node Node to check against
+   * @param expected_latency Latency incurred
+   * @param tolerance Acceptable percentage threshold/tolerance
+   */
+  void check_max_latency(int node, unsigned int expected_latency, double tolerance) {
+      double latency = max_node_latency[node - 1];
+      if (latency < expected_latency) {
+        BOOST_CHECK_CLOSE(latency, expected_latency, tolerance);
+      }
+  }
+
+private:
+  static test_utils::CassSessionPtr session_;
+  const cql::cql_ccm_bridge_configuration_t& configuration_;
+  test_utils::CassClusterPtr cluster_;
+  static bool is_running_;
+#if UV_VERSION_MAJOR == 0
+  static uv_loop_t* loop_;
```

Comment:
loop is used after the single thread is created in start_query_execution in start_thread here; https://github.com/datastax/cpp-driver/pull/153/files#diff-20b2a3bbb7595403a32ccb337f67af91R139. Is this implemented wrong?